bundles/paikkatietoikkuna/coordinateconversion/components/select.js

```diff
@@ -0,0 +1,168 @@
+                        dropdowns[i].find('.'+ instances[2].getValue()).show();
+                    } 
+                    if( instances[2].getValue() === "GK" ) {
+                        dropdowns[3].find('option').hide();
+                        dropdowns[i].find('.'+ instances[2].getValue()).show();
+                    }
+                    if( instances[3].getValue() !== "COORDSYS_DEFAULT" ) {
+                        me.instance.startingSystem = true;
+                    } else {
+                        me.instance.startingSystem = true;
+                    }
+                    if( instances[4].getValue() === "KORKEUSJ_DEFAULT") {
+                        rows.each( function ( idx, row ) {
+                            var lastCell = jQuery(this).find('td:nth-last-child(2)');
+                            if( !lastCell.hasClass('heightsystem' )) {
+                                lastCell.addClass('heightsystem');
+                            }
+                        });
+                    } else {
+                        rows.each( function ( idx, row ) {
+                            var lastCell = jQuery(this).find('td:nth-last-child(2)');
+                            lastCell.attr("contenteditable", false);
+                            lastCell.removeClass('heightsystem');
+                        })
+                    }
+                        
+                    if( instances[1].getValue() === 'KOORDINAATISTO_SUORAK_2D' ) {
+                        jQuery('.map-projection').show();
+                    } else {
+                        jQuery('.map-projection').hide();
+                        instances[2].resetToPlaceholder();
+                    }
+
+                    if( instances[0].getValue() !== me.instance.currentDatum ||me.instance.currentDatum === undefined ) {
+                        if( i !== 0 ) {
+                            instances[i].resetToPlaceholder();
+                        }
+                    }
+                    if( instances[1].getValue() !== me.instance.currentCoordinatesystem ||me.instance.currentCoordinatesystem === undefined ) {
+                        if( i > 1) {
+                            instances[i].resetToPlaceholder();
+                        }
+                    }
+                    values = [];
+                    instances[i].update();
+                    for ( var j = 0; j < instances.length; j++ ) {
+                        var vl = instances[j].getValue();
+                        values.push( vl );
+                    }
+                    if( i == instances.length -1 ) {
+                        if( this.id === 'inputcoordsystem' ) {
+                            me.instance.inputTable.updateTitle( values );
+                            me.instance.inputTable.isEditable( me.instance.clipboardInsert );
+                        } else if( this.id === 'targetcoordsystem' ) {
+                            me.instance.outputTable.updateTitle( values );
```

Comment:
You should make this component observable and bind listeners outside of it for these kinds of things. Instead of triggering functions with deep reference structure outside this component.